model_templates/gpu_nemo_textgen/custom_model/custom.py

```diff
@@ -1,25 +1,106 @@
+import logging
 import os
+import subprocess
+from pathlib import Path
+from subprocess import CalledProcessError
+
 import boto3
 from mlpiper.extra.aws_helper import AwsHelper
-from datarobot_drum import RuntimeParameters
 
+from datarobot_drum import RuntimeParameters
+from datarobot_drum.custom_task_interfaces.user_secrets import SecretType
+from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.gpu_predictors import get_optional_parameter
 
 # this path is required by NeMo Inference Server
 NEMO_MODEL_STORE_DIR = "/model-store"
 
 
 def load_model(code_dir: str):
-    print("Downloading model from S3...")
+    load_model_response = (
+        "succeeded"  # a non-empty response is required to signal that load_model succeeded
+    )
+
+    s3_url = get_optional_parameter("s3Url")
```

Comment:
This is a model-template and thus should define the param appropriately and then it is an actual error if the `get()` raises.

Reply:
Agree. I added two different schemes:
- for NGC registry
https://github.com/datarobot/datarobot-user-models/blob/882835887839fe98a3b988400577b82254b02a46/model_templates/gpu_nemo_textgen/custom_model/model-metadata.yaml#L5-L13
- for S3 (used by our functional tests)
https://github.com/datarobot/datarobot-user-models/blob/4596dc0335c9ba3fc18714f62ee7e5faab9ccc85/tests/fixtures/gpu_nemo_textgen/custom_model/model-metadata.yaml#L5-L13

Reply:
This is how the `custom.py` looks like:
https://github.com/datarobot/datarobot-user-models/blob/882835887839fe98a3b988400577b82254b02a46/model_templates/gpu_nemo_textgen/custom_model/custom.py#L5-L10